### ISSUE TITLE
dev server with Cordova : cannot configure HTTPS

### ISSUE THREAD

**Issue by mpacary** [quasarframework/quasar-cli#194]
Quoting [this forum post](https://forum.quasar-framework.org/topic/2279/https-dev-server-with-cordova) from [Narmer23](https://forum.quasar-framework.org/user/narmer23):

-----------------------------------

The Quasar devServer can be started in an HTTPS enviroment setting the https attribute in **quasar.conf.js**:
```
devServer: {
  https: true
 [...]
}
```
Unfortunately this setting is ignored when starting the devServer in Cordova or Electron mode, since false is hard-coded in **<project>/node_modules/quasar-cli/lib/quasar-config.js** line 528:
```
  else if (this.ctx.mode.cordova || this.ctx.mode.electron) {
    Object.assign(cfg.devServer, {
      https: false,
      open: false
    })
  }
```
Changing here the https value to true “solves” this issue, but being inside the library it’s not a good way to solve it. We should have the possibility to configure it from the project’s **quasar.conf.js**.

-----------------------------

Reproduced using Quasar 0.16.4.

**Comment by rstoenescu** [comment]
Hi,

There's a reason why it's disabled by default, otherwise your Cordova or Electron would fail to load due to technical issues on behalf of those two platforms. If things have changed in the meantime and you can confirm 100% that the issues are no longer valid, please **do reopen this ticket**. Thank you.

**Comment by mpacary** [comment]
@rstoenescu Thanks for your quick reply & explanations. I close the issue created on the **quasar-cli** repo immediately.

**Comment by insign** [comment]
Should not these considerations be in the documentation?

**Comment by TheCheat** [comment]
@rstoenescu perhaps then, its just better to warn while building, rather than strictly prohibiting from using https: true with electron/cordova, that this _might_ cause some errors, 'cause im not getting any and rewriting this setting in node_modules dir isn't best solution, IMO.

**Comment by rstoenescu** [comment]
Allowing you to run https on the devserver would translate as: "Quasar CLI allows you to fail miserably when deploying". The reason is simple. The devserver runs on `http://` protocol while the app on a mobile phone (built and installed with an .apk or .app) runs on `file://` protocol. Different protocols, different browser security settings for each protocol. You might be happy that your features might work while developing your app, but when you actually deploy it, big surprises may arise. So Quasar CLI helps you not fall into this trap.

**Comment by TheCheat** [comment]
> What about this issue? Hacking in node_modules doesn't seem to be an appropriate solution... I want to use location stuff in my app which requires a "secure origin"

Same reason. I cant use WebRTC on devServer with non-secure origin.

**Comment by tonydoesathing** [comment]
@mpacary 's workaround worked for me. I needed the https so I could make use of MediaDevices.getUserMedia() for camera access, which is disabled unless coming from https (it works when you build the app for production and are no longer running the dev server, but it's a hassle to rebuild and deploy the app while I'm debugging and writing the app)

Not sure about a better solution though other than changing it to a warning
Thank you all for your efforts and insights with this!

**Comment by rstoenescu** [comment]
quasar-cli v0.17.24 (not yet released, most likely tomorrow) will allow HTTPS as you requested.

**Comment by rstoenescu** [comment]
Released quasar-cli v0.17.24.
